backends/imgui_impl_sdlrenderer.cpp

```diff
@@ -0,0 +1,223 @@
+	ImVec2 clip_off = draw_data->DisplayPos;         // (0,0) unless using multi-viewports
+	ImVec2 clip_scale = renderScale;
+
+    ImGui_ImplSDLRenderer_SetupRenderState();
+
+    for (int n = 0; n < draw_data->CmdListsCount; n++) {
+
+        const ImDrawList* cmd_list = draw_data->CmdLists[n];
+        const ImDrawVert* vtx_buffer = cmd_list->VtxBuffer.Data;
+        const ImDrawIdx* idx_buffer = cmd_list->IdxBuffer.Data;
+        
+        for (int cmd_i = 0; cmd_i < cmd_list->CmdBuffer.Size; cmd_i++)
+        {
+            const ImDrawCmd* pcmd = &cmd_list->CmdBuffer[cmd_i];
+            if (pcmd->UserCallback)
+            {
+                // User callback, registered via ImDrawList::AddCallback()
+                // (ImDrawCallback_ResetRenderState is a special callback value used by the user to request the renderer to reset render state.)
+                if (pcmd->UserCallback == ImDrawCallback_ResetRenderState) {
+                    ImGui_ImplSDLRenderer_SetupRenderState();
+                } else { 
+                    pcmd->UserCallback(cmd_list, pcmd);
+                }
+            }
+            else
+            {
+                // Project scissor/clipping rectangles into framebuffer space
+                ImVec4 clip_rect;
+                clip_rect.x = (pcmd->ClipRect.x - clip_off.x) * clip_scale.x;
+                clip_rect.y = (pcmd->ClipRect.y - clip_off.y) * clip_scale.y;
+                clip_rect.z = (pcmd->ClipRect.z - clip_off.x) * clip_scale.x;
+                clip_rect.w = (pcmd->ClipRect.w - clip_off.y) * clip_scale.y;
+
+                if (clip_rect.x < fb_width && clip_rect.y < fb_height && clip_rect.z >= 0.0f && clip_rect.w >= 0.0f)
+                {
+                    SDL_Rect r;
+                    r.x = clip_rect.x;
+                    r.y = clip_rect.y;
+                    r.w = clip_rect.z - clip_rect.x;
+                    r.h = clip_rect.w - clip_rect.y;
+
+                    SDL_RenderSetClipRect(bd->SDLRenderer, &r);
+
+
+                    int xy_stride = sizeof(ImDrawVert);
+                    float *xy = (float *)((char*)vtx_buffer + IM_OFFSETOF(ImDrawVert, pos));
+
+                    int uv_stride = sizeof(ImDrawVert);
+                    float *uv = (float*)((char*)vtx_buffer + IM_OFFSETOF(ImDrawVert, uv));
+
+                    int col_stride = sizeof(ImDrawVert);
+                    int *color = (int*)((char*)vtx_buffer + IM_OFFSETOF(ImDrawVert, col));
+
+					SDL_Texture *tex = (SDL_Texture*)pcmd->TextureId;
+
+                    SDL_RenderGeometryRaw(bd->SDLRenderer, tex, 
+                            xy, xy_stride, color,
+                            col_stride, 
+                            uv, uv_stride, 
+                            cmd_list->VtxBuffer.Size,
+                            idx_buffer, pcmd->ElemCount, sizeof (ImDrawIdx));
+
+                }
+            }
+            idx_buffer += pcmd->ElemCount;
+        }
+    }
```

Comment:
Should render clipping be disabled at the end of the ImGui_ImplSDLRenderer_RenderDrawData()? 
i.e., `SDL_RenderSetClipRect(bd->SDLRenderer, NULL);`

Any additional render calls with that SDL_Renderer made by the user will only be able to be seen when the imgui windows are over the area in which the user has drawn.